File: app/src/main/java/com/hippo/ehviewer/dao/FilterDao.java
package com.hippo.ehviewer.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface FilterDao extends BasicDao<Filter> {
    @Query("SELECT * FROM FILTER")
    List<Filter> list();

    @Update
    void update(Filter filter);

    @Insert
    long insert(Filter filter);

    @Delete
    void delete(Filter filter);

    @Query("SELECT * FROM FILTER")
    List<Filter> fakeList();

    @Insert
    void fakeInsert(Filter t);
}

File: app/src/main/java/com/hippo/ehviewer/dao/FilterDao.kt
package com.hippo.ehviewer.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface FilterDao {
    @Query("SELECT * FROM FILTER")
    suspend fun list(): List<Filter>

    @Update
    suspend fun update(filter: Filter)

    @Insert
    suspend fun insert(t: Filter): Long

    @Insert
    suspend fun insert(filters: List<Filter>)

    @Delete
    suspend fun delete(filter: Filter)

    @Query("SELECT * FROM FILTER WHERE TEXT = :text AND MODE = :mode")
    suspend fun load(text: String, mode: Int): Filter?
}
